### PR TITLE
refactor(migrate): add `createDatabase` & `ensureConnectionValidity` to RPC

### PR DESCRIPTION
**Pull request by Jolg42**
- Removed `planMigration`
- Reordered all definitions alphabetically
- added `createDatabase` & `ensureConnectionValidity` to RPC API

Related https://github.com/prisma/team-orm/issues/46

TODO in next PR
- [ ] Replace usage of command in `createDatabase` to `createDatabase` RPC
- [ ] Replace usage of command in `canConnectToDatabase` to `ensureConnectionValidity` RPC

### CODE REVIEW

**Comment by Jolg42** [comment]
TODO: fix conflicts

**Review by Druue — APPROVED** [review]

File: packages/migrate/src/MigrateEngine.ts [hunk 1]
```diff
@@ -51,78 +52,171 @@ export class MigrateEngine {
...
-    args: EngineArgs.ListMigrationDirectoriesInput,
-  ): Promise<EngineResults.ListMigrationDirectoriesOutput> {
-    return this.runCommand(this.getRPCPayload('listMigrationDirectories', args))
+
+  /**
+   * Create the logical database from the Prisma schema.
+   */
+  public createDatabase(args: EngineArgs.CreateDatabaseInput): Promise<EngineResults.CreateDatabaseOutput> {
+    return this.runCommand(this.getRPCPayload('createDatabase', args))
   }
```
Comment (jkomyno) [inline packages/migrate/src/MigrateEngine.ts:75]: This is a new method: https://prisma.github.io/prisma-engines/doc/migration_core/json_rpc/index.html#-createdatabase

File: packages/migrate/src/MigrateEngine.ts [hunk 2]
```diff
@@ -51,78 +52,171 @@ export class MigrateEngine {
...
-  public planMigration(args: EngineArgs.PlanMigrationInput): Promise<EngineResults.PlanMigrationOutput> {
-    return this.runCommand(this.getRPCPayload('planMigration', args))
+
+  /**
+   * Make sure the migration engine can connect to the database from the Prisma schema.
+   */
+  public ensureConnectionValidity(args: EngineArgs.EnsureConnectionValidityInput): Promise<void> {
+    return this.runCommand(this.getRPCPayload('ensureConnectionValidity', args))
   }
```
Comment (jkomyno) [inline packages/migrate/src/MigrateEngine.ts:128]: This is a new method: https://prisma.github.io/prisma-engines/doc/migration_core/json_rpc/index.html#-ensureconnectionvalidity

File: packages/migrate/src/MigrateEngine.ts [hunk 3]
```diff
@@ -51,78 +52,171 @@ export class MigrateEngine {
...
-  public applyMigrations(args: EngineArgs.ApplyMigrationsInput): Promise<EngineResults.ApplyMigrationsOutput> {
-    return this.runCommand(this.getRPCPayload('applyMigrations', args))
+
+  /**
+   * Compares two databases schemas from two arbitrary sources,
+   * and display the difference as either a human-readable summary,
+   * or an executable script that can be passed to dbExecute.
+   * Connection to a shadow database is only necessary when either the from or the to params is a migrations directory.
+   * Diffs have a direction. Which source is from and which is to matters.
+   * The resulting diff should be thought as a migration from the schema in from to the schema in to.
```
Comment (jkomyno) [inline packages/migrate/src/MigrateEngine.ts:151]:
```suggestion
   * The resulting diff should be thought of as a migration from the schema in `args.from` to the schema in `args.to`.
```

Comment (Jolg42) [reply packages/migrate/src/MigrateEngine.ts]: Note: this was taken from the Rust doc

**Review by jkomyno — APPROVED** [review]

**Pull request merged**